zerver/views/messages.py

```diff
@@ -319,7 +340,9 @@ def highlight_string_text_offsets(text, locs):
     pos = 0
     result = u''
     for loc in locs:
-        (offset, length) = loc
+        (offset, length) = highlight_match_tag_pos(string, pos, loc, u'<', u'>')
+        if offset < pos:
+            continue
```

Comment:
@timabbott The case is when the `highlight_match_tag_pos` detects that we are inside a tag, and moves the length forward to match all of it -- then we might end up with a location in locs which is also in the same tag (mainly with pgroonga), and it is better to make sure we don't accidentally add some text twice.